app/models/casebook/contents.rb

```diff
@@ -0,0 +1,102 @@
+# == Schema Information
+#
+# Table name: casebooks
+#
+#  id            :integer          not null, primary key
+#  title         :string
+#  slug          :string
+#  subtitle      :string
+#  headnote      :text
+#  public        :boolean          default(TRUE), not null
+#  book_id       :integer
+#  ordinals      :integer          default([]), not null, is an Array
+#  copy_of_id    :integer
+#  is_alias      :boolean
+#  resource_type :string
+#  resource_id   :integer
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#
+
+class Casebook::Contents < Casebook::Generic
+  default_scope {where.not(book_id: nil)}
+
+  validates_length_of :ordinals, minimum: 1
+  validates_uniqueness_of :ordinals, scope: :book_id, if: :book_id?, message: 'are not unique'
+
+  before_validation :move_siblings, if: :ordinals_changed?
```

Comment:
Yep, AR. `attribute_changed?` is true if `attribute` has a different value from the persisted value (up to when the `save` callbacks fire).